File: app/snippets/docs/integration/libraries/java/basic.java
import com.surrealdb.connection.SurrealWebSocketConnection;
import com.surrealdb.driver.SyncSurrealDriver;

import java.util.List;
import java.util.Map;

public class App
{
    public static void main( String[] args )
    {
        SurrealWebSocketConnection conn = new SurrealWebSocketConnection("localhost", 8000, false);
        conn.connect(5);
        SyncSurrealDriver driver = new SyncSurrealDriver(conn);
        driver.signIn("root", "root");
        driver.use("namespace-name", "database-name");
        String tableName = "user";
        driver.delete(tableName);
        User tobie = driver.create(tableName, User.builder().name("Tobie").build());
        User jaime = driver.create(tableName, User.builder().name("Tobie").build());
        List<Map<String, String>> updates = driver.update(jaime.id, Map.of(
                "name", "Jaime"));
        List<User> allUsers = driver.select(tableName, User.class);
        System.out.printf("All users = %s", allUsers);
        conn.disconnect();
    }
}

File: app/snippets/docs/integration/libraries/nodejs/basic.js
const { default: Surreal } = require('surrealdb.js');

const db = new Surreal('http://127.0.0.1:8000/rpc');

async function main() {

	try {

		// Signin as a namespace, database, or root user
		await db.signin({
			user: 'root',
			pass: 'root',
		});

		// Select a specific namespace / database
		await db.use({ns: 'test', db: 'test'});

		// Create a new person with a random id
		let created = await db.create("person", {
			title: 'Founder & CEO',
			name: {
				first: 'Tobie',
				last: 'Morgan Hitchcock',
			},
			marketing: true,
			identifier: Math.random().toString(36).slice(2, 12),
		});

		// Update a person record with a specific id
		let updated = await db.merge("person:jaime", {
			marketing: true,
		});

		// Select all people records
		let people = await db.select("person");

		// Perform a custom advanced query
		let groups = await db.query('SELECT marketing, count() FROM type::table($tb) GROUP BY marketing', {
			tb: 'person',
		});

	} catch (e) {

		console.error('ERROR', e);

	}

}

main();
